s3/replication/build-runalltest.sh

```diff
@@ -0,0 +1,192 @@
+
+# Active an environment
+python3 -m venv $REPLICATION_DIR/s3env
+source $REPLICATION_DIR/s3env/bin/activate
+
+# Check and download dependencies
+pip3 install --upgrade setuptools
+
+pip3 install -r $REPLICATION_DIR/common/requirements.txt
+pip3 install -r $REPLICATION_DIR/manager/requirements.txt
+pip3 install -r $REPLICATION_DIR/replicator/requirements.txt
+
+cd $REPLICATION_DIR
+make clean install
+
+# Create and copy files
+S3_IN_MEMRORY_SVC_FILE=$S3_REPLICATION_COMMON_SRC_DIR/systemd/s3_in_memory.service
+MANGER_SVC_FILE=$S3_REPLICATION_MANAGER_SRC_DIR/systemd/s3replicationmanager.service
+REPLICATOR_SVC_FILE=$S3_REPLICATOR_SRC_DIR/systemd/s3replicator.service
+
+# Systemd service files
+SYSD_S3_IN_MEMORY_SVC_FILE=/etc/systemd/system/s3_in_memory.service
+SYSD_REPLICATION_MANAGER_SVC_FILE=/etc/systemd/system/s3replicationmanager.service 
+SYSD_REPLICATOR_SVC_FILE=/etc/systemd/system/s3replicator.service
+
+# Create creadential file in home directory with below path.
+# ~/.cortxs3/credentials.yaml
+CORTX_S3_CRED_FILE=~/.cortxs3/credentials.yaml
+CORTX_S3_DIR="$(dirname "$CORTX_S3_CRED_FILE")"
+if [ -f "$CORTX_S3_CRED_FILE" ]; then
+    echo "$CORTX_S3_CRED_FILE exist!"
+else
+    if [ -d $CORTX_S3_DIR]; then
+        echo "$CORTX_S3_DIR exist!"
+    else
+        mkdir $CORTX_S3_DIR
+    fi
+    touch $CORTX_S3_CRED_FILE
+    echo "access_key: '<your access_key>'" > $CORTX_S3_CRED_FILE
+    echo "secret_key: '<your secret_key>'" >> $CORTX_S3_CRED_FILE
+fi
+
+rm -f $SYSD_S3_IN_MEMORY_SVC_FILE 
+rm -f $SYSD_REPLICATION_MANAGER_SVC_FILE
+rm -f $SYSD_REPLICATOR_SVC_FILE
+
+# Strip hardcoded lines
+head -n -2 $S3_IN_MEMRORY_SVC_FILE > $SYSD_S3_IN_MEMORY_SVC_FILE
+head -n -2 $MANGER_SVC_FILE > $SYSD_REPLICATION_MANAGER_SVC_FILE
+head -n -2 $REPLICATOR_SVC_FILE > $SYSD_REPLICATOR_SVC_FILE
+
+# Setting environment and working directory in service files
+Environment="PYTHONPATH=$S3_REPLICATION_COMMON_SRC_DIR"
+echo "Environment=$Environment" >> $SYSD_S3_IN_MEMORY_SVC_FILE
+echo "Environment=$Environment" >> $SYSD_REPLICATION_MANAGER_SVC_FILE
+echo "Environment=$Environment" >> $SYSD_REPLICATOR_SVC_FILE
+
+echo "WorkingDirectory=$S3_REPLICATION_COMMON_SRC_DIR" >> $SYSD_S3_IN_MEMORY_SVC_FILE
+echo "ExecStart=/usr/bin/python3 $S3_REPLICATION_COMMON_TEST_DIR/s3_in_memory.py" >> $SYSD_S3_IN_MEMORY_SVC_FILE
+
+echo "WorkingDirectory=$S3_REPLICATION_MANAGER_SRC_DIR" >> $SYSD_REPLICATION_MANAGER_SVC_FILE
+echo "WorkingDirectory=$S3_REPLICATOR_SRC_DIR" >> $SYSD_REPLICATOR_SVC_FILE
+
+# Start daemon process
+DAEMON_SVC_DIR=/etc/systemd/system 
+
+echo "Service files created at : "$DAEMON_SVC_DIR
+
+sudo systemctl daemon-reload
+
+systemctl enable s3_in_memory.service
+systemctl enable s3replicationmanager.service
+systemctl enable s3replicator.service
+
+# Starting services
+systemctl start s3_in_memory.service
+svc_status=$(systemctl is-active s3_in_memory.service)
+
+if [ $svc_status != "active" ]
```

Comment:
may be helper stop_all_services() method can reduce if-else ladder and make it more readable.

Reply:
done